contracts/src/IncredibleSquaringServiceManager.sol

```diff
@@ -25,13 +25,14 @@ contract IncredibleSquaringServiceManager is ServiceManagerBase {
     }
 
     constructor(
-        IDelegationManager _delegationManager,
+        IAVSDirectory _avsDirectory,
         IRegistryCoordinator _registryCoordinator,
         IStakeRegistry _stakeRegistry,
         IIncredibleSquaringTaskManager _incredibleSquaringTaskManager
     )
         ServiceManagerBase(
-            _delegationManager,
+            _avsDirectory,
+            IPaymentCoordinator(address(0)), // inc-sq doesn't need to deal with payments
```

Comment:
This does not exist on the `testnet-holesky` branch - which im assuming is what we're trying to reference end-to-end.

Reply:
Pointing at testnet-holesky makes sense, just need to be clearly stated in README. 
In general, you might want to think, what's the purpose of this project. If that an example for developers to write own ones, probably you need to put an effort to keep it up to date with a dev branch. It's that's more an example how AVS might work, keeping on not very old, well define point should be fine.

Reply:
Dev moves too fast I think; it will deviate from the docs fairly quickly.
Personally think this repo/example should align with M2; and hence `testnet-holesky` for both `eigenlayer-middleware` and `eigenlayer-contracts`.

Currently it's a mess reading the docs, then this repo, and looking at code for other AVS's.
Too much complexity to get simple example running - and then finding out that this example is outdated lol.

Reply:
Apologies guys, I've been in your situation many times with other protocol's repos... and now it's my turn to be on this side of the heat. 100% agree with your critiques, just not super sure how to best deal with them. In general would you guys prefer if this always pointing to the contracts on holesky, so that you could deploy there? Or would you prefer it was on latest changes always, given that this is meant to get a look into the future of protocol changes?

The mess we're in is that we're using foundry which uses submodules, so middleware -> core contracts. So it's very hard to for eg get the middleware (avs) contract latest changes while keep pointing at the core contracts deployed on testnet/mainnet. In this precise case I needed some fixes on the deployment scripts which are unfortunately only on the latest dev head, which then forced me to include the payments changes :\

Reply:
I think maybe we just replicate what eigenlayer-contracts and eigenlayer-middleware are doing. 
I.e. have a `testnet-holesky` branch that references the same branch of both the upstream dependencies. 
Then have a dev branch (could be main/master) which aligns with the middleware and core contract Dev branches. 

As for myself I am, trying to follow official docs to first deploy an AVS to testnet, then hopefully to mainnet (near future).
Was trying to adapt the linear squaring example but seeming to be more difficult than I'd initially hoped.

Reply:
@samlaf - there is no critique, just observations (many eyes of open source). And I agree with your point that the usage of submodules makes things messier. I had 3 terminal windows next to each other to ensured that I saw what dependencies were pointing at what ;-)

Reply:
Points well taken and heard! Synced with @8sunyuan , check his message at the bottom. I'll resolve this and merge for now, but please do keep us in check!